### DISCUSSION TITLE
UniGetUI 3.1.2 beta 3

### DISCUSSION THREAD

**Discussion by marticliment** [Devolutions/UniGetUI#2785] [Announcements]
![](https://img.shields.io/github/downloads/marticliment/wingetui/3.1.2-beta2/total?style=for-the-badge&color=red)

~~NOTE: This beta is expected to be the latest beta before release.
NOTE: This release is not digitally signed for convenience. Stable releases will always be signed.~~
This was expected to be the last beta release before stable, yet stable will be delayed due to issues with the code signing certificate.
I have been told that the issues should be fixed in a week or two, as of 08/10/2024 (dd/mm/aa)

# Changelog
 - UI improvements
   - The interface is smoother. All background tasks are run threaded instead of using the asynchronous interface
   - Package lists will maintain selection when the packages change (up to the possible extent)
   - Improvements to the "Discover Packages" page behaviour
   - Filters panel has a slide-out/slide-in animation
   - Better messages when a UAC prompt has been canceled

 - Package managers improvements
   - Cargo will be properly detected
   - NPM packages will be entirely parsed via json, improving reliability
   - Pip and Scoop package details parsers have undergone important improvements
   - Chocolatey authenticated sources will now show on UniGetUI
   - WinGet COM API will work when running UniGetUI as administrator. Bundled WinGet package detection has been improved
   - WinGet PowerShell module will not be used anymore to parse packages. 

<hr /><em>This discussion was created from the release <a href='https://github.com/marticliment/UniGetUI/releases/tag/3.1.2-beta3'>UniGetUI 3.1.2 beta 3</a>.</em>

**Comment by phoenixlucky** [comment]
I guess the certificate thing is gonna be harder than I thought.

**Reply by marticliment** [reply]
Yes, and I am sorry for that.
I will be publishing the stable release as soon as I can codesign again

**Reply by LaKanDoR** [reply]
Do you use a self-signed certificate?

openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 365 -nodes

openssl dgst -sha256 -sign key.pem -out signature.bin code.exe

or

signtool sign /f cert.pem /p your_password /tr http://timestamp.digicert.com code.exe

**Reply by marticliment** [reply]
No, self-signed certificates do not fullfill the security requirements for which I began signing UniGetUI

**Reply by LaKanDoR** [reply]
thank you for the information. 
Sometimes it takes some time until the signing got valid under Windows.
